pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -268,20 +268,30 @@ func (c *Cluster) Start() error {
 		return errors.Wrapf(err, "failed to update ceph storage status")
 	}
 
-	if c.replaceOSD != nil {
-		delOpts := &k8sutil.DeleteOptions{MustDelete: true, WaitOptions: k8sutil.WaitOptions{Wait: true}}
-		err := k8sutil.DeleteConfigMap(c.clusterInfo.Context, c.context.Clientset, OSDReplaceConfigName, namespace, delOpts)
-		if err != nil {
-			return errors.Wrapf(err, "failed to delete the %q configmap", OSDReplaceConfigName)
+	if c.spec.Storage.Store.UpdateStore == OSDStoreUpdateConfirmation {
+		if c.replaceOSD != nil {
```

Comment:
```suggestion
	if c.spec.Storage.Store.UpdateStore == OSDStoreUpdateConfirmation && c.replaceOSD != nil {
```

can we do this?

Reply:
Nope. The other checks inside the main if block are not related to `c.replaceOSD != nil` condition.

Reply:
ohh I see it now, thanks